include/deal.II/fe/fe_base.h

```diff
@@ -18,1046 +18,6 @@
-                                     0 :
-                                     face_no];
-}
-
-template <int dim>
-unsigned int
-FiniteElementData<dim>::get_first_face_quad_index(
-  const unsigned int face_no) const
-{
-  return first_quad_index_of_faces[first_quad_index_of_faces.size() == 1 ?
-                                     0 :
-                                     face_no];
-}
-
-template <int dim>
-internal::GenericDoFsPerObject
-internal::GenericDoFsPerObject::generate(const FiniteElementData<dim> &fe)
-{
-  const auto reference_cell = fe.reference_cell();
-
-  internal::GenericDoFsPerObject result;
-
-  result.dofs_per_object_exclusive.resize(4);
-  result.dofs_per_object_inclusive.resize(4);
-  result.object_index.resize(4);
-
-  unsigned int counter = 0;
-
-  for (unsigned int v : reference_cell.vertex_indices())
-    {
-      const auto c = fe.template n_dofs_per_object<0>(v);
-
-      result.dofs_per_object_exclusive[0].emplace_back(c);
-      result.dofs_per_object_inclusive[0].emplace_back(c);
-      result.object_index[0].emplace_back(counter);
-
-      counter += c;
-    }
-
-  if (dim >= 2)
-    for (unsigned int l : reference_cell.line_indices())
-      {
-        const auto c = fe.template n_dofs_per_object<1>(l);
-
-        result.dofs_per_object_exclusive[1].emplace_back(c);
-        result.dofs_per_object_inclusive[1].emplace_back(
-          c + 2 * fe.template n_dofs_per_object<0>());
-        result.object_index[1].emplace_back(counter);
-
-        counter += c;
-      }
-
-  if (dim == 3)
-    for (unsigned int f : reference_cell.face_indices())
-      {
-        const auto c = fe.template n_dofs_per_object<2>(f);
-
-        result.dofs_per_object_exclusive[2].emplace_back(c);
-        result.dofs_per_object_inclusive[2].emplace_back(fe.n_dofs_per_face(f));
-        result.object_index[2].emplace_back(counter);
-
-        counter += c;
-      }
-
-  {
-    const auto c = fe.template n_dofs_per_object<dim>();
-
-    result.dofs_per_object_exclusive[dim].emplace_back(c);
-    result.dofs_per_object_inclusive[dim].emplace_back(fe.n_dofs_per_cell());
-    result.object_index[dim].emplace_back(counter);
-
-    counter += c;
-  }
-
-  for (unsigned int d = dim + 1; d <= 3; ++d)
-    {
-      result.dofs_per_object_exclusive[d].emplace_back(0);
-      result.dofs_per_object_inclusive[d].emplace_back(0);
-      result.object_index[d].emplace_back(counter);
-    }
-
-  result.first_object_index_on_face.resize(3);
-  for (unsigned int face_no : reference_cell.face_indices())
-    {
-      result.first_object_index_on_face[0].emplace_back(0);
-
-      result.first_object_index_on_face[1].emplace_back(
-        fe.get_first_face_line_index(face_no));
-
-      result.first_object_index_on_face[2].emplace_back(
-        fe.get_first_face_quad_index(face_no));
-    }
-
-  return result;
-}
-
-
-#endif // DOXYGEN
-
-
-DEAL_II_NAMESPACE_CLOSE
+DEAL_II_WARNING("This file is deprecated. Simply use <deal.II/fe/fe_data.h>.")
```

Comment:
@drwells This is already part of this PR.

Reply:
there it is!